### ISSUE TITLE
image_weights

### ISSUE THREAD

**Issue by shining-love** [ultralytics/ultralytics#19453]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

First of all, thank you for your excellent work. Secondly, I would like to know how to set the image_weights parameter in this project using v8, v5, v11, etc., to control the long-tail distribution of data (data imbalance). In previous projects like YOLOv5, there was an image_weights hyperparameter, but I couldn't find where to set and use image_weights in this repository. I look forward to your reply.

**Comment by xuan-xuan6** [comment]
> 👋 Hello [@shining-love](https://github.com/shining-love), thank you for your interest in Ultralytics 🚀! We appreciate your insightful question regarding the `image_weights` parameter.
> 
> For your inquiry:
> 
> * If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. This will make it easier for us to assist you effectively.
> * If this is a ❓ Question regarding custom training or usage, kindly share additional details to provide more context, such as logs, configurations, or relevant examples.
> 
> ## Resources
> We recommend exploring our [Docs](https://docs.ultralytics.com/) for comprehensive guidance. You might find the following sections particularly helpful:
> 
> * [Python Usage](https://docs.ultralytics.com/usage/python/)
> * [CLI Usage](https://docs.ultralytics.com/usage/cli/)
> * [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)
> 
> Feel free to also join the Ultralytics community for engagement and support:
> 
> * Discuss your queries on our [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time chat.
> * Connect on [Discourse](https://community.ultralytics.com/) for in-depth discussions.
> * Join the conversation on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge.
> 
> ## Upgrade and Verify
> Make sure you're using the latest `ultralytics` package to ensure access to the latest features and bug fixes. Upgrade your environment as follows:
> 
> pip install -U ultralytics
> Ensure your Python environment meets the requirements ([**Python>=3.8**](https://www.python.org/) and [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/)).
> 
> ## Environments
> YOLO models are tested in the following environments. You can try your workflow in these for streamlined performance:
> 
> * **Notebooks with free GPU**: [![Run on Gradient](https://camo.githubusercontent.com/6df71b7d7e0b09a2e97776f416bcd40acf48fc87337f8bcce48e4235537daf1f/68747470733a2f2f6173736574732e706170657273706163652e696f2f696d672f6772616469656e742d62616467652e737667)](https://console.paperspace.com/github/ultralytics/ultralytics) [![Open In Colab](https://camo.githubusercontent.com/96889048f8a9014fdeba2a891f97150c6aac6e723f5190236b10215a97ed41f3/68747470733a2f2f636f6c61622e72657365617263682e676f6f676c652e636f6d2f6173736574732f636f6c61622d62616467652e737667)](https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb) [![Open In Kaggle](https://camo.githubusercontent.com/c7135949c5c6882489e68f4af05a78a759460a4db256b86df3097e04419b4d9e/68747470733a2f2f6b6167676c652e636f6d2f7374617469632f696d616765732f6f70656e2d696e2d6b6167676c652e737667)](https://www.kaggle.com/models/ultralytics/yolo11)
> * **Google Cloud** Deep Learning VM: See the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
> * **Amazon** Deep Learning AMI: Follow the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
> * **Docker Image**: Refer to the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) [![Docker Pulls](https://camo.githubusercontent.com/6dcbe0a245fadcfd5202d971a92467bd2fcab86efebb2228d73468152c680573/68747470733a2f2f696d672e736869656c64732e696f2f646f636b65722f70756c6c732f756c7472616c79746963732f756c7472616c79746963733f6c6f676f3d646f636b6572)](https://hub.docker.com/r/ultralytics/ultralytics)
> 
> ## Status
> [![Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg)](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule)
> 
> If the CI badge above is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are running smoothly. CI tests validate the behavior of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across multiple platforms.
> 
> 🔧 Thank you for your patience! Please note that this is an automated response. An Ultralytics engineer will review your question and respond with further assistance soon.

I just want to know how to set the image_weights parameter. I saw someone providing a training script with this parameter in another question, such as in this link（https://github.com/ultralytics/ultralytics/issues/7419）. However, I couldn't find where this parameter is set and whether it is currently supported. Is it possible that image_weights is a custom addition made by someone else? I'm quite confused about this.

**Comment by Y-T-G** [comment]
You can't set weights for images in Ultralytics

**Comment by shining-love** [comment]
> You can't set weights for images in Ultralytics

Hello, how can I implement the image_weights function? I remember that this function was available in previous versions. Why was this function removed in the new version?

**Comment by Y-T-G** [comment]
I don't think it was present in Ultralytics. It is present in the YOLOv5 repo. But that's a different project and codebase.

**Comment by shining-love** [comment]
> I don't think it was present in Ultralytics. It is present in the YOLOv5 repo. But that's a different project and codebase.

OK, thank you for your reply. I'd like to ask again, why was the image_weights function removed in the new version? This function is very friendly to the long-tail distribution. It's really a pity that this function was removed in the new version.

**Comment by Y-T-G** [comment]
Ultralytics was built ground up from scratch. It wasn't removed. It was simply not added.

If you want, you can use a workaround like here:
https://github.com/ultralytics/ultralytics/issues/15044#issuecomment-2276387717

Or use a weighted dataloader:
https://y-t-g.github.io/tutorials/yolo-class-balancing/

**Comment by Y-T-G** [comment]
Duplicate of https://github.com/ultralytics/ultralytics/issues/21082
